cypress/e2e/tests/pages/extensions-compatibility-tests/elemental.spec.ts

```diff
@@ -0,0 +1,214 @@
+
+    // click on install button on card
+    extensionsPo.extensionCardInstallClick(EXTENSION_NAME);
+    extensionsPo.extensionInstallModal().should('be.visible');
+
+    // select version and click install
+    extensionsPo.installModalSelectVersionLabel(EXTENSION_VERSION);
+    extensionsPo.installModalInstallClick();
+
+    // let's check the extension reload banner and reload the page
+    extensionsPo.extensionReloadBanner().should('be.visible');
+    extensionsPo.extensionReloadClick();
+
+    // make sure extension card is in the installed tab
+    extensionsPo.extensionTabInstalledClick();
+    extensionsPo.extensionCardClick(EXTENSION_NAME);
+    extensionsPo.extensionDetailsTitle().should('contain', EXTENSION_NAME);
+    extensionsPo.extensionDetailsCloseClick();
+  });
+
+  it('Should setup all of the needed backend parts', () => {
+    cy.intercept('POST', 'v1/catalog.cattle.io.clusterrepos/rancher-charts?action=install').as(EXTENSION_CHART_CREATION);
+
+    elementalPo.goTo();
+    elementalPo.waitForTitle('h1', 'OS Management');
+
+    elementalPo.installOperatorBtnClick();
+
+    elementalPo.waitForInstallChartPage('rancher-charts', 'elemental');
+
+    // we need to change the namespace picker in order for the install check on the list view
+    namespacePicker.toggle();
+    namespacePicker.clickOptionByLabel('All Namespaces');
+    namespacePicker.closeDropdown();
+
+    elementalPo.chartInstallNext();
+    elementalPo.chartInstallClick();
+    elementalPo.chartInstallWaitForInstallationAndCloseTerminal(EXTENSION_CHART_CREATION, ['elemental-operator-crds', 'elemental-operator']);
+
+    elementalPo.goTo();
+    elementalPo.waitForTitle('[data-testid="elemental-main-title"]', 'OS Management Dashboard');
+  });
+
+  it('Should create an Elemental registration endpoint', () => {
+    cy.intercept('POST', 'v1/elemental.cattle.io.machineregistrations/fleet-default').as('machineRegCreation');
+
+    elementalPo.goTo();
+    elementalPo.sideMenuNavTo('Registration Endpoints');
+    elementalPo.createClick();
+
+    elementalPo.genericNameInput().set(REG_ENDPOINT_NAME);
+    elementalPo.genericYamlEditor().value().then((val) => {
+      // convert yaml into json to update values
+      const json: any = jsyaml.load(val);
+
+      json.config.elemental.install.device = REG_ENDPOINT_DEVICE_PATH;
+
+      elementalPo.genericYamlEditor().set(jsyaml.dump(json));
+      elementalPo.saveOrCreateResource().click();
+
+      cy.wait('@machineRegCreation', { requestTimeout: 15000 }).then(({ response }) => {
+        expect(response?.statusCode).to.eq(201);
+        expect(response?.body.metadata).to.have.property('name', REG_ENDPOINT_NAME);
+        expect(response?.body.spec.config.elemental.install).to.have.property('device', REG_ENDPOINT_DEVICE_PATH);
+      });
+    });
+  });
+
+  it('Should create an Elemental resource via YAML (Inventory of Machines)', () => {
+    function poolingSchemaDefinition() {
```

Comment:
this should have a max try counter, then fail if exceeds some value. otherwise it could continue indefinitely

Reply:
I'll give it a big number, since I've seen this stuck here for a while